service/history/execution/mutable_state_builder.go

```diff
@@ -4508,8 +4511,14 @@ func (e *mutableStateBuilder) startTransactionHandleDecisionFailover(
 		)}
 	}
 
-	lastWriteSourceCluster := e.clusterMetadata.ClusterNameForFailoverVersion(lastWriteVersion)
-	currentVersionCluster := e.clusterMetadata.ClusterNameForFailoverVersion(currentVersion)
+	lastWriteSourceCluster, err := e.clusterMetadata.ClusterNameForFailoverVersion(lastWriteVersion)
+	if err != nil {
+		return false, err
```

Comment:
since I'm guessing these will all be hard to tell apart in logs: probably worth adding some descriptive wrapping (and it's safe to wrap here regardless of existing error checks, because this is a new category of errors for this code)